zipkin-lens/src/components/TracePage/TraceSummaryHeader.jsx

```diff
@@ -105,6 +106,81 @@ const TraceSummaryHeader = React.memo(({ traceSummary, rootSpanIndex }) => {
       ? config.logsUrl.replace(/{traceId}/g, traceSummary.traceId)
       : undefined;
 
+  const archivePostUrl =
+    config.archivePostUrl && traceSummary ? config.archivePostUrl : undefined;
+
+  const archiveUrl =
+    config.archiveUrl && traceSummary
+      ? config.archiveUrl.replace('{traceId}', traceSummary.traceId)
+      : undefined;
+
+  const { enqueueSnackbar } = useSnackbar();
+
+  const archiveClick = useCallback(() => {
+    const notify = function notify(message, variant) {
+      enqueueSnackbar(message, {
+        variant,
+        anchorOrigin: {
+          vertical: 'top',
+          horizontal: 'center',
+        },
+        autoHideDuration: 10000, // 10 seconds
+      });
+    };
+
+    // We don't store the raw json in the browser yet, so we need to make an
+    // HTTP call to retrieve it again.
+    fetch(`${api.TRACE}/${traceSummary.traceId}`)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error('Failed to fetch trace from backend');
+        }
+        return response.json();
+      })
+      .then((json) => {
+        // Add zipkin.archived tag to root span
+        /* eslint-disable-next-line no-restricted-syntax */
+        for (const span of json) {
+          if ('parentId' in span === false) {
+            const tags = span.tags || {};
+            tags['zipkin.archived'] = 'true';
+            span.tags = tags;
+            break;
+          }
+        }
+
+        fetch(archivePostUrl, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(json),
+        })
+          .then((response) => {
+            if (
+              !response.ok ||
+              (response.status !== 202 && response.status === 200)
+            ) {
+              throw new Error('Failed to archive the trace');
+            }
+            if (archiveUrl) {
+              notify(
+                `Archive successful! This trace is now accessible at ${archiveUrl}`,
+                'success',
+              );
+            } else {
+              notify(`Archive successful!`, 'success');
+            }
+          })
+          .catch((error) => {
+            notify(`Failed to archive the trace: ${error}`, 'error');
+          });
+      })
+      .catch(() => {
+        notify('Failed to fetch trace from backend', 'error');
```

Comment:
If you use the same pattern as above of catching the error and notifying in the notication, you should be able to have a single `catch` instead of the nested